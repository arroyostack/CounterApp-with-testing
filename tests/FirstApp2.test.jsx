import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


describe( 'Test in FirstApp.jsx', () => {
  const title = 'Mecano en tus muerto';

  test( 'should match the snapshot', () => {
    const { container } = render( <FirstApp title={ title } /> );

    expect( container ).toMatchSnapshot();
  } );

  test( 'should show mesage "Mecano en tus muerto"', () => {
    render( <FirstApp title={ title } /> );

    expect( screen.getByText( title ) ).toBeTruthy();
  } );

  test( 'should show title in h1', () => {
    render( <FirstApp title={ title } /> );

    expect( screen.getByRole( 'heading', { level: 1 } ).innerHTML ).toContain( title );
  } );


} );