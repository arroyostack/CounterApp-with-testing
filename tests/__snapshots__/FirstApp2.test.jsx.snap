// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp.jsx should match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Mecano en tus muerto
  </h1>
  <p>
    There is no subtitle
  </p>
  <p>
    There is no name
  </p>
</div>
`;
